lib/src/Uncertainty/Algorithm/Simulation/QuadrantSampling.cxx

```diff
@@ -0,0 +1,209 @@
+  quadrantIndex_ = quadrantIndex;
+}
+
+UnsignedInteger QuadrantSampling::getQuadrantIndex() const
+{
+  return quadrantIndex_;
+}
+
+void QuadrantSampling::setQuadrantOrientation(const NumericalPoint & quadrantOrientation)
+{
+  if ((quadrantOrientation.getDimension() > 0) && (quadrantOrientation.getDimension() != getDimension())) throw InvalidDimensionException(HERE) << "Quadrant orientation dimension (" << quadrantOrientation.getDimension() << " ) should be " << getDimension();
+  quadrantOrientation_ = quadrantOrientation;
+  updateRotation();
+}
+
+NumericalPoint QuadrantSampling::getQuadrantOrientation() const
+{
+  return quadrantOrientation_;
+}
+
+void QuadrantSampling::setStrataIndices(const Indices & strataIndices)
+{
+  UnsignedInteger m = (1 << strataIndices.getSize());
+  if (quadrantIndex_ >= m) throw InvalidArgumentException(HERE) << "Quadrant index (" << quadrantIndex_ << " ) should be < " << m;
+  if (strataIndices.getSize() > samplingStrategy_.getDimension()) throw InvalidDimensionException(HERE) << "Strata indices (" << strataIndices.getSize() << ") should be < " << samplingStrategy_.getDimension();
+  stataIndices_ = strataIndices;
+  updateRotation();
+}
+
+
+Indices QuadrantSampling::getStrataIndices() const
+{
+  return stataIndices_;
+}
+
+/* String converter */
+String QuadrantSampling::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << QuadrantSampling::GetClassName()
+      << " derived from " << SamplingStrategyImplementation::__repr__();
+  return oss;
+}
+
+
+void QuadrantSampling::updateRotation()
+{
+
+  if ( quadrantOrientation_.getDimension() > 0 )
+  {
+    UnsignedInteger p = getStrataIndices().getSize();
+
+    NumericalPoint u(p);
+    for ( UnsignedInteger j = 0; j < p; ++ j )
+    {
+      u[j] = quadrantOrientation_[stataIndices_[j]];
+    }
+
+    // Gram-Schmidt algorithm
+    NumericalSample f(p, p);
+    // f0 = u* / ||u*||
+    f[0] = u / u.norm();
+    for ( UnsignedInteger k = 1; k < p; ++ k )
+    {
+      // fk = ek
+      f[k][k] = 1.0;
+      // fk -= SUM <ek, fi> fi
+      for ( UnsignedInteger i = 0; i < k; ++ i )
+      {
+        f[k] -= dot(f[k], f[i]) * f[i];
+      }
+      // fk = fk / ||fk||
+      f[k] /= NumericalPoint(f[k]).norm();
+    }
+
+    // H is the transformation E -> F, columns = fk
+    SquareMatrix H(p);
+    for ( UnsignedInteger i = 0; i < p; ++ i )
+      for ( UnsignedInteger j = 0; j < p; ++ j )
+        H(j, i) = f[i][j];
+
+    // P is the transformation G -> F
+    SquareMatrix P(p);
+    for ( UnsignedInteger j = 0; j < p; ++ j )
+    {
+      P(0, j) = 1. / sqrt(1.0 * p);
+    }
+    for ( UnsignedInteger k = 1; k < p; ++ k )
+    {
+      P(k, k) = (p - k) / sqrt(1.0 * (p - k) * (p - k + 1));
+      for ( UnsignedInteger i = 0; i < p; ++ i )
+      {
+        if ((i < 1) || (i > k))
+          P(k, i) -= 1. / sqrt(1.0 * (p - k) * (p - k + 1));
+      }
+    }
+
+    // R is the tranformation E -> G
+    R_ = P * H;
```

Comment:
In fact if I use the constructor based on raw contents it already transposes it as NS and Matrix have opposite storage orders.